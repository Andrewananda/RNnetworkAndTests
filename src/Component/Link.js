import React from 'react';
import {View} from 'react-native';
import {WebView} from 'react-native-webview';

export default function Link(props) {
  return (
    <View>
      <WebView source={{uri: props.page}} />
    </View>
  );
}
