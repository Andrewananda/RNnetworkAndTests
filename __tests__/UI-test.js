import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import Link from '../src/Component/Link';

it('renders link', () => {
  const tree = renderer
    .create(<Link page="http://www.instagram.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
