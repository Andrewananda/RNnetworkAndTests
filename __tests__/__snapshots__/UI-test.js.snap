// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "http://www.instagram.com",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
